crates/steel-core/src/steel_vm/primitives.rs

```diff
@@ -671,24 +674,51 @@ fn not(value: &SteelVal) -> bool {
     matches!(value, SteelVal::BoolV(false))
 }
 
+#[steel_derive::function(name = "number?", constant = true)]
+pub fn numberp(value: &SteelVal) -> bool {
+    matches!(
+        value,
+        SteelVal::IntV(_)
+            | SteelVal::BigNum(_)
+            | SteelVal::FractV(_)
+            | SteelVal::BigFract(_)
+            | SteelVal::NumV(_)
+    )
+}
+
 #[steel_derive::function(name = "int?", constant = true)]
 fn intp(value: &SteelVal) -> bool {
-    matches!(value, SteelVal::IntV(_))
+    matches!(value, SteelVal::IntV(_) | SteelVal::BigNum(_))
 }
 
 #[steel_derive::function(name = "integer?", constant = true)]
 fn integerp(value: &SteelVal) -> bool {
-    matches!(value, SteelVal::IntV(_))
+    intp(value)
 }
 
 #[steel_derive::function(name = "float?", constant = true)]
 fn floatp(value: &SteelVal) -> bool {
     matches!(value, SteelVal::NumV(_))
 }
 
-#[steel_derive::function(name = "number?", constant = true)]
-fn numberp(value: &SteelVal) -> bool {
-    matches!(value, SteelVal::NumV(_) | SteelVal::IntV(_))
+#[steel_derive::function(name = "real?", constant = true)]
+fn realp(value: &SteelVal) -> bool {
+    matches!(
+        value,
+        SteelVal::IntV(_)
```

Comment:
I'm thinking of some renames to match the r7rs language more. Should be really easy with `rust-analyzer`. LMK which ones you think are worth doing.

```
IntV -> Int
BigNum -> BigInt
FractV -> Rational or maybe Rat
BigFractV -> BigRational or maybe BigRat
NumV -> Real
```

Reply:
I'm okay with the changes, Rational and BigRational are fine. My only concern is that with the enum being public the rename is a breaking change. I'm fine with the breaking change but we'd just have to bump the version numbers since its still pre 1.0

Reply:
Ack. Updated just `Fract` since that's the one that's annoying me the most.